ProcessOptimizer/utils/utils.py

```diff
@@ -1,886 +1,896 @@
+            )
+
+            # Error message.
+            msg = """All dimensions must be instances of the Dimension-class,
+                  but found: {}"""
+            msg = msg.format(err_dims)
+            raise ValueError(msg)
+
+        # Ensure all dimensions have names.
+        if any(dim.name is None for dim in dimensions):
+            # List of the dimensions that have no names.
+            err_dims = list(filter(lambda dim: dim.name is None, dimensions))
+
+            # Error message.
+            msg = "All dimensions must have names, but found: {}"
+            msg = msg.format(err_dims)
+            raise ValueError(msg)
+
+        @wraps(func)
+        def wrapper(x):
+            """
+            This is the code that will be executed every time the
+            wrapped / decorated `func` is being called.
+            It takes `x` as a single list of parameters and
+            converts them to named arguments and calls `func` with them.
+
+            Parameters
+            ----------
+            * `x` [list]:
+                A single list of parameters e.g. `[123, 3.0, 'linear']`
+                which will be converted to named arguments and passed
+                to `func`.
+
+            Returns
+            -------
+            * `objective_value`
+                The objective value returned by `func`.
+            """
+
+            # Ensure the number of dimensions match
+            # the number of parameters in the list x.
+            if len(x) != len(dimensions):
+                msg = (
+                    "Mismatch in number of search-space dimensions. "
+                    "len(dimensions)=={} and len(x)=={}"
+                )
+                msg = msg.format(len(dimensions), len(x))
+                raise ValueError(msg)
+
+            # Create a dict where the keys are the names of the dimensions
+            # and the values are taken from the list of parameters x.
+            arg_dict = {dim.name: value for dim, value in zip(dimensions, x)}
+
+            # Call the wrapped objective function with the named arguments.
+            objective_value = func(**arg_dict)
+
+            return objective_value
+
+        return wrapper
+
+    return decorator
+
+
+def y_coverage(res, return_plot=False, random_state=None, horizontal=False):
+    """
+    A function to calculate the expected range of observable function values
+    given a result instans. This can be compared with the actual observed
+    range of function values.
+    """
+    assert len(res.func_vals) != 0, "train model before using this function"
+    observed_min = res.func_vals.min()
+    observed_max = res.func_vals.max()
+    min_x, expected_min = expected_minimum(
+        res, n_random_starts=20, random_state=None, minmax="min"
+    )
+    max_x, expected_max = expected_minimum(
+        res, n_random_starts=20, random_state=None, minmax="max"
+    )
```

Comment:
```suggestion
    max_x, expected_max = expected_minimum(res, random_state=random_state, minmax="max")
```
Remove calls of default parameters.